src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetApplicationOverviewHandler.cs

```diff
@@ -0,0 +1,267 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Http;
+using SFA.DAS.AdminService.Web.Infrastructure;
+using SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway;
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.AdminService.Web.Domain;
+
+namespace SFA.DAS.AdminService.Web.Handlers.Gateway
+{
+    public class GetApplicationOverviewHandler : IRequestHandler<GetApplicationOverviewRequest, RoatpGatewayApplicationViewModel>
+    {
+        private readonly IRoatpApplicationApiClient _applyApiClient;
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly IHttpContextAccessor _contextAccessor;
+
+
+        public GetApplicationOverviewHandler(IRoatpApplicationApiClient applyApiClient, IQnaApiClient qnaApiClient, IHttpContextAccessor contextAccessor)
+        {
+            _applyApiClient = applyApiClient;
+            _qnaApiClient = qnaApiClient;
+            _contextAccessor = contextAccessor;
+        }
+        public async Task<RoatpGatewayApplicationViewModel> Handle(GetApplicationOverviewRequest request, CancellationToken cancellationToken)
+        {
+            var application = await _applyApiClient.GetApplication(request.ApplicationId);
+            if (application is null)
+            {
+                return null;
+            }
+
+            // Setting Application Data => TODO: To be stored in session.
```

Comment:
Why storing in the session? Are you definitely sure this will never change?

Reply:
It's a comment only. I still think that if store some common  information we will save on some repetitive API calls to get the application data.